warehouse/warehouse.go

```diff
@@ -684,6 +698,12 @@ func (wh *HandleT) mainLoop(ctx context.Context) {
 		wh.configSubscriberLock.RLock()
 		wg := sync.WaitGroup{}
 		wg.Add(len(wh.warehouses))
+
+		// We will be measuring the overall lag
+		// as part of the process.
+		whSchedulerLag := stats.DefaultStats.NewStat("wh_scheduler.total_scheduling_time", stats.TimerType)
```

Comment:
Not sure whether it is going to provide the scheduler lag. Since creation is happening in a different GoRoutine. Here we are just iterating over the warehouses and trying to create jobs in a different GoRoutine.